be/src/env/env_hdfs.cpp

```diff
@@ -97,8 +101,53 @@ Status HdfsRandomAccessFile::readv_at(uint64_t offset, const Slice* res, size_t
     return Status::InternalError("HdfsRandomAccessFile::readv_at not implement");
 }
 
-Status HdfsRandomAccessFile::size(uint64_t* size) const {
+// ===================================  S3RandomAccessFile  =========================================
+
+S3RandomAccessFile::S3RandomAccessFile(S3Client* client, const std::string& bucket, const std::string& object,
+                                       size_t object_size)
+        : _client(client), _bucket(bucket), _object(object), _object_size(object_size) {
+    _file_name = "s3://" + _bucket + "/" + _object;
+}
+
+Status S3RandomAccessFile::read(uint64_t offset, Slice* res) const {
+    size_t read_size = 0;
+    Status st = _client->get_object_range(_bucket, _object, offset, res->size, res->data, &read_size);
+    if (!st.ok()) return st;
+    res->size = read_size;
+    return st;
+}
+
+Status S3RandomAccessFile::read_at(uint64_t offset, const Slice& res) const {
+    size_t read_size = 0;
+    RETURN_IF_ERROR(_client->get_object_range(_bucket, _object, offset, res.size, res.data, &read_size));
+    if (read_size != res.size) {
+        return Status::InternalError(
+                strings::Substitute("fail to read enough data, file=$0, offset=$1, size=$2, expect=$3", _file_name,
+                                    offset, read_size, res.size));
+    }
+    return Status::OK();
+}
+
+Status S3RandomAccessFile::readv_at(uint64_t offset, const Slice* res, size_t res_cnt) const {
     // TODO: implement
-    return Status::InternalError("HdfsRandomAccessFile::size not implement");
+    return Status::InternalError("S3RandomAccessFile::readv_at not implement");
+}
+
+std::shared_ptr<RandomAccessFile> create_random_access_hdfs_file(const HdfsFsHandle& handle,
```

Comment:
may be we need a more concrete name

Reply:
I think it's okay for now. Do you have any suggestion?